packages/waku/src/lib/renderers/html-renderer.ts

```diff
@@ -102,6 +104,16 @@ globalThis.__WAKU_PREFETCHED__ = {
       if (scriptsClosed) {
         return;
       }
+      if (!mainJsSent) {
+        // enqueue `mainJs` first in the body to avoid the document.body undefined error, before we used to inject it in the head which sometimes made it load before the document.body was loaded
```

Comment:
My oversight, this should be DEV only, right?

`npm run examples:prd:04_promise`

<img width="894" alt="image" src="https://github.com/dai-shi/waku/assets/490574/1c55ff4f-134f-4de8-bec7-4bc12ce5e143">

Reply:
This should in general send the mainJs one time, weird error, do you know why that happens?

Reply:
Here's the fix #528 

`main.tsx` doesn't exist on PRD, right?

Reply:
Yes, what exists then if that does not exist? because as far as I checked the website, there was an alternative. And we need that to be in the body too, so the behaviour is normalized between two environments.

Reply:
Oh, ok the reason we get 404 is that we give a path that's not valid for prd! And in prd we give a different value, correct?
if yes, and we control that value, are we sure it'd be passed in the body rather than the head? Or are we sure that the body is going to be returned? so we don't get the error.

Reply:
See #528 for the fix. Yes, it's already in the head in PRD. It's already done in the build step, so we don't need to do anything in the runtime process.

Reply:
But when it's in the head, and we use document.body there, are we sure that document.body is going to be always defined?

Reply:
https://github.com/dai-shi/waku/pull/518#issuecomment-1962238483

Reply:
But the error that this PR fixed was reproduced with async too, so I don't think that can always be true and we need to be sure about that.

Reply:
My understanding is that the issue this PR fixed is for HMR in dev mode. I never see such an issue in PRD.

Reply:
Yes it fixed the HMR issue after the refresh in DEV, but I was just concerned how it'd be handled in PRD. That's it. sorry for the confusion.

Reply:
Yep, they're in the head! because the body is always sent in the initial template (index.html)